### PR TITLE
EIP1-4574: EIP1:4664: Add (OPTIONAL) delivery information for create AED document endpoint

### PR DESCRIPTION
**Pull request by vishalgupta-valtech**
ℹ️ Delivery information will be made mandatory in subsequent PRs . Task [EIP1-4668](https://technologyprogramme.atlassian.net/browse/EIP1-4668)  captures this

### CODE REVIEW

File: src/main/resources/openapi/PrintAPIs.yaml [hunk 1]
```diff
@@ -1094,6 +1142,9 @@ components:
           example: 01234 567890
         address:
           $ref: '#/components/schemas/Address'
+        delivery:
+          # TODO make delivery as mandatory (EIP1-4668) once changes in VCA (EIP1-4665) are completed. TODO
+          $ref: '#/components/schemas/CertificateDelivery'
```
Comment (vishalgupta-valtech) [inline src/main/resources/openapi/PrintAPIs.yaml:1147]: Currently vca-api calls this endpoint to generate AED, therefore initially making it as an optional. Subsequent subtasks are captured in the story to eventually make this field as mandatory once VCA populates this.

Comment (vishalgupta-valtech) [reply src/main/resources/openapi/PrintAPIs.yaml:1147]: Currently `vca-api` calls this endpoint to `generate AED`, therefore to not break vca-api, we are initially marking this new attribute as an optional field. 
Subsequent subtasks are captured in the story to eventually make this as `mandatory` field once VCA populates this in [PR](https://github.com/cabinetoffice/eip-ero-voter-card-applications-api/pull/575) .

File: src/test/kotlin/uk/gov/dluhc/printapi/testsupport/testdata/dto/GenerateAnonymousElectorDocumentDtoBuilder.kt [hunk 2]
```diff
@@ -56,7 +56,7 @@ fun buildGenerateAnonymousElectorDocumentDto(
     )
 
 fun buildValidAddressDto(
-    fakeAddress: net.datafaker.Address = DataFaker.faker.address(),
+    fakeAddress: net.datafaker.providers.base.Address = DataFaker.faker.address(),
```
Comment (vishalgupta-valtech) [inline src/test/kotlin/uk/gov/dluhc/printapi/testsupport/testdata/dto/GenerateAnonymousElectorDocumentDtoBuilder.kt:59]: Had to change this due to `updated` datafaker version.

File: src/main/kotlin/uk/gov/dluhc/printapi/mapper/AnonymousElectorDocumentMapper.kt [hunk 3]
```diff
@@ -26,6 +27,11 @@ abstract class AnonymousElectorDocumentMapper {
     @Autowired
     protected lateinit var clock: Clock
 
+    abstract fun toGenerateAnonymousElectorDocumentDto(
```
[the remark this answers is not in the document]
Comment (Neil-Massey) [reply src/main/kotlin/uk/gov/dluhc/printapi/mapper/AnonymousElectorDocumentMapper.kt:30]: I think it was in a separate one as this is the mapper for `AnonymousElectorDocument` (not `GenerateAnonymousElectorDocumentDto`)

File: build.gradle.kts [hunk 4]
```diff
@@ -102,15 +102,15 @@ dependencies {
...
 
-    testImplementation("org.testcontainers:testcontainers:1.17.5")
-    testImplementation("org.testcontainers:mysql:1.17.5")
+    testImplementation("org.testcontainers:testcontainers:1.17.6")
+    testImplementation("org.testcontainers:mysql:1.17.6")
 
-    testImplementation("com.github.tomakehurst:wiremock-jre8:2.34.0")
-    testImplementation("net.datafaker:datafaker:1.6.0")
+    testImplementation("com.github.tomakehurst:wiremock-jre8:2.35.0")
+    testImplementation("net.datafaker:datafaker:1.8.0")
```
Comment (Neil-Massey) [inline build.gradle.kts:113]: Nice - good to keep these advancing 👍

File: build.gradle.kts [hunk 5]
```diff
@@ -230,7 +230,7 @@ tasks.withType<BootBuildImage> {
     environment = mapOf("BP_HEALTH_CHECKER_ENABLED" to "true")
     buildpacks = listOf(
         "urn:cnb:builder:paketo-buildpacks/java",
-        "gcr.io/paketo-buildpacks/health-checker",
+        "gcr.io/paketo-buildpacks/health-checker"
```
Comment (Neil-Massey) [inline build.gradle.kts:233]: I think the trailing `,` was done deliberately so when additional lines are added the PR diff is slightly simpler.

**Review by Neil-Massey — APPROVED** [review]
LGTM - Few minor observations/comments

File: build.gradle.kts (hunk 5 above)
Comment (vishalgupta-valtech) [reply build.gradle.kts:233]: For some reason the build failed for ktlint 11.3.0, so had to remove them . Let me put that back on for ktlint `11.0.0`

File: src/main/kotlin/uk/gov/dluhc/printapi/mapper/AnonymousElectorDocumentMapper.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (vishalgupta-valtech) [reply src/main/kotlin/uk/gov/dluhc/printapi/mapper/AnonymousElectorDocumentMapper.kt:30]: Yeah agree, I wanted to club the dto and entity for `AnonymousElectorDocument` within same mapper

**Pull request merged**